python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py

```diff
@@ -51,11 +55,49 @@ def default() -> "TemplatedValueResolver":
     def with_context(self, **additional_context) -> "TemplatedValueResolver":
         return TemplatedValueResolver(context={**self.context, **additional_context})
 
-    def resolve(self, val: Any) -> Any:
+    def _resolve_value(self, val: Any) -> Any:
         return NativeTemplate(val).render(**self.context) if isinstance(val, str) else val
 
+    def _resolve(
+        self,
+        val: Any,
+        valpath: Optional[Sequence[Union[str, int]]],
+        should_render: Callable[[Sequence[Union[str, int]]], bool],
```

Comment:
I'm not following this `should_render` stuff. Under what conditions will this be true versus false?

Reply:
I'm still finding this code path quite confusing given what is happening here should be pretty straightforward?

Reply:
Updated the names in the PR and added some comments, but basically there's just two options for the should_render function:

1. a function that always returns true (this is what translators and the like use -- they will greedily render any values they come across)
2. a function that inspects the json schema of the target params type to determine if we're deferring or not (this is what we use in the pre-processing stage to keep some fields unrendered)